components/certmanager/prepare_apiserver_patch.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash -eu
+#
+# Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+pushd "$KUBIFY_STATE_PATH" 1> /dev/null
+master_count=$(terraform output master_count)
+ip=$(terraform output master_ips)
+bastion=$(terraform output bastion)
+pem=$KUBIFY_STATE_PATH/gen/nodes_privatekey.pem
+popd 1> /dev/null
+
+doSSH() {
+    # do ssh, ignore unknown hosts, don't save hosts, suppress warnings about saving hosts
+    ssh -i $pem -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o "ProxyCommand=ssh -i $pem -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -W %h:%p ubuntu@$bastion" core@$ip "$@" 2> >(grep -v "Warning: Permanently added .* to the list of known hosts\.")
+}
+
+if [ $# -gt 0 ] && [ $1 == "undo" ] ; then
```

Comment:
you don't check for one master node like below

Reply:
Yes and no. The `terraform output` command needs to be run from the folder containing the terraform state, so I guess it extracts the information from there. So, the number should be correct as long as 1. the caller has the up-to-date terraform state and 2. nobody has modified the infrastructure manually without using terraform for it. But I agree that in this case it might be better to get the actual number of master nodes from kubernetes directly.